src/arch/riscv/arch.c

```diff
@@ -103,15 +103,17 @@ void arch_print_regs(struct nk_regs * r) {
 
 void * arch_read_sp(void) {
     void * sp = NULL;
-    __asm__ __volatile__ ( "mv %[_r], sp" : [_r] "=r" (rsp) : : "memory" );
+    __asm__ __volatile__ ( "mv %[_r], sp" : [_r] "=r" (sp) : : "memory" );
     return sp;
 }
 
 void arch_relax(void) {
-    asm volatile ("pause");
+    // RISCV HACK
+    // asm volatile ("pause");
```

Comment:
There isn't a riscv equiv to the x86 pause instruction, so this isn't much of a hack.